### ISSUE TITLE
Why the regression branch output 81 classes' location (not 80 classes) in old version of mmdetection?

### ISSUE THREAD

**Issue by Icecream-blue-sky** [open-mmlab/mmdetection#6942]
I am reproducing a 19-year paper, and the mmdetection it uses is an old version. It's based on Faster R-CNN. And I find that in old verson of mmdetection, the regression branch of bbox_head outputs 81 classes' location (not 80 classes in COCO dataset) when reg_class_agnostic is False. The code is below:
```python
mmdet/models/bbox_heads/bbox_head.py
@HEADS.register_module
class BBoxHead(nn.Module):
    """Simplest RoI head, with only two fc layers for classification and
    regression respectively"""

    def __init__(self,
                 with_avg_pool=False,
                 with_cls=True,
                 with_reg=True,
                 roi_feat_size=7,
                 in_channels=256,
                 num_classes=81,
                 target_means=[0., 0., 0., 0.],
                 target_stds=[0.1, 0.1, 0.2, 0.2],
                 reg_class_agnostic=False):
        super(BBoxHead, self).__init__()
        assert with_cls or with_reg
        self.with_avg_pool = with_avg_pool
        self.with_cls = with_cls
        self.with_reg = with_reg
        self.roi_feat_size = roi_feat_size
        self.in_channels = in_channels
        self.num_classes = num_classes
        self.target_means = target_means
        self.target_stds = target_stds
        self.reg_class_agnostic = reg_class_agnostic

        in_channels = self.in_channels
        if self.with_avg_pool:
            self.avg_pool = nn.AvgPool2d(roi_feat_size)
        else:
            in_channels *= (self.roi_feat_size * self.roi_feat_size)
        # num_classes is set as 81
        **if self.with_cls:
            self.fc_cls = nn.Linear(in_channels, num_classes)**
        **if self.with_reg:
            out_dim_reg = 4 if reg_class_agnostic else 4 * num_classes
            self.fc_reg = nn.Linear(in_channels, out_dim_reg)**
        self.debug_imgs = None
```
But in the latest version of mmdetection, the reg branch outputs 80 classses' location:
```python
if self.with_cls:
            # need to add background class
            if self.custom_cls_channels:
                cls_channels = self.loss_cls.get_cls_channels(self.num_classes)
            else:
                cls_channels = num_classes + 1
            self.fc_cls = build_linear_layer(
                self.cls_predictor_cfg,
                in_features=in_channels,
                out_features=cls_channels)
        if self.with_reg:
            out_dim_reg = 4 if reg_class_agnostic else 4 * num_classes
            self.fc_reg = build_linear_layer(
                self.reg_predictor_cfg,
                in_features=in_channels,
                out_features=out_dim_reg)
```
I want to know whether these two settings have any effect on the performance of the model, which is very important to me. Besides, I can't understand why the old version need to predict the position of the background class (81). It has no meaning and is not supervised by loss function.

**Comment by RangiLyu** [comment]
This is a bug in the previous version of mmdet.
